api/environments/identities/models.py

```diff
@@ -1,18 +1,29 @@
 import hashlib
 import typing
 
+import boto3
+from django.conf import settings
 from django.db import models
 from django.db.models import Prefetch, Q
 from django.utils.encoding import python_2_unicode_compatible
+from django_lifecycle import AFTER_CREATE, LifecycleModel, hook
+from flag_engine.identities.builders import build_identity_dict
 
 from environments.identities.traits.models import Trait
 from environments.models import Environment
 from features.models import FeatureState
 from features.multivariate.models import MultivariateFeatureStateValue
 
+# Intialize the dynamo client globally
+dynamo_identity_table = None
+if settings.IDENTITIES_TABLE_NAME_DYNAMO:
+    dynamo_identity_table = boto3.resource("dynamodb").Table(
+        settings.IDENTITIES_TABLE_NAME_DYNAMO
+    )
+
 
 @python_2_unicode_compatible
-class Identity(models.Model):
+class Identity(LifecycleModel):
```

Comment:
@matthewelwell  I feel like there must be a better way to handle this, but we have to move this to global api anyway

Reply:
Not sure I follow, sorry?

Reply:
A better way to handle the whole identity document update. right now we are updating/creating the document at three places and I am not sure if we are missing any endpoint.
But I guess it will become more streamlined with
```
No, we will need to do that using hooks on the django models eventually (and overriding the bulk_create method...). But I'd like to do it asyncronously as soon as possible so it needs to be totally decoupled from the API request.
```

Reply:
Yeah we are missing some e.g: when user creates a trait using /trait endpoint

Reply:
Surely that's covered by the hook you added on the Trait model. I think the only thing that we're missing regarding traits is the bulk create endpoint? Which I think you can do by just overriding the `bulk_create` on the model manager.